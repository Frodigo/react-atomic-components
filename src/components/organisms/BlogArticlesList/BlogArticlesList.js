import React from 'react';
import PropTypes from 'prop-types';

import BlogPost from '../../molecues/BlogPost';

import classes from './BlogArticlesList.module.css';

const BlogArticlesList = (props) => {
    const { items } = props;

    return (
        <div className={classes.list}>
            {items.map((post) => (
                <BlogPost
                    title={post.title}
                    author={post.author}
                    date={post.date}
                    url={post.url}
                    image={post.image}
                    shortDescription={post.shortDescription}
                    key={post.id}
                />
            ))}
        </div>
    );
};

BlogArticlesList.propTypes = {
    items: PropTypes.array.isRequired
};

export default BlogArticlesList;
