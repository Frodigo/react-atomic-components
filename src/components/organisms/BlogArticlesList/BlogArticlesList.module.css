.list {
    border-top: solid 2px var(--color-text);
    margin-top: 15px;
}