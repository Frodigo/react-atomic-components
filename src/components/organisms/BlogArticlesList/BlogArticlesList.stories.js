import React from 'react';

import { default as BlogArticlesListComponent } from './BlogArticlesList';

export default {
    title: 'Organisms/BlogArticlesList',
    component: BlogArticlesListComponent
};

const Template = (args) => <BlogArticlesListComponent {...args} />;

export const Default = Template.bind({});
Default.args = {
    items: [
        {
            id: 1,
            title: 'Title 1',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        },
        {
            id: 2,
            title: 'Title 2',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        },
        {
            id: 3,
            title: 'Title 3',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        },
        {
            id: 4,
            title: 'Title 4',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        }
    ]
};
