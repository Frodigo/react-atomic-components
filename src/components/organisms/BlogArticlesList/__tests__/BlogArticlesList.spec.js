import React from 'react';
import { render } from '@testing-library/react';

import BlogArticlesList from '../BlogArticlesList';

describe('Blog articles list should be rendered', () => {
    const mockedItems = [
        {
            id: 1,
            title: 'Title 1',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        },
        {
            id: 2,
            title: 'Title 2',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        },
        {
            id: 3,
            title: 'Title 3',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        },
        {
            id: 4,
            title: 'Title 4',
            author: 'Jan Kowalski',
            date: '23 December, 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription: 'lorem upsum'
        }
    ];

    test('Articles list renders correctly', () => {
        const { container } = render(<BlogArticlesList items={mockedItems} />);

        expect(container).toMatchSnapshot();
    });
});
