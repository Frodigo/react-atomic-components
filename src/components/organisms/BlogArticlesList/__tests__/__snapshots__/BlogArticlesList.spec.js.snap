// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog articles list should be rendered Articles list renders correctly 1`] = `
<div>
  <div
    class="list"
  >
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Title 1
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Jan Kowalski
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1608678000000"
              >
                23 December 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            lorem upsum
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Title 1"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Title 2
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Jan Kowalski
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1608678000000"
              >
                23 December 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            lorem upsum
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Title 2"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Title 3
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Jan Kowalski
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1608678000000"
              >
                23 December 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            lorem upsum
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Title 3"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Title 4
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Jan Kowalski
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1608678000000"
              >
                23 December 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            lorem upsum
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Title 4"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
  </div>
</div>
`;
